kuma/wiki/tests/test_views_code.py

```diff
@@ -0,0 +1,141 @@
+from waffle.models import Switch
+import pytest
+
+from kuma.attachments.models import Attachment
+from kuma.attachments.tests import make_test_file
+from kuma.core.urlresolvers import reverse
+from kuma.wiki.models import Revision
+
+from . import normalize_html
+
+
+@pytest.fixture
+def code_sample_doc(root_doc, wiki_user):
+    sample_page = """
+        <p>This is a page. Deal with it.</p>
+        <div id="sample1" class="code-sample">
+            <pre class="brush: html">Some HTML</pre>
+            <pre class="brush: css">.some-css { color: red; }</pre>
+            <pre class="brush: js">window.alert("HI THERE")</pre>
+        </div>
+        <p>test</p>
+        {{ EmbedLiveSample('sample1') }}
+    """
+    root_doc.current_revision = Revision.objects.create(
+        document=root_doc, creator=wiki_user, content=sample_page)
+    root_doc.save()
+    return root_doc
+
+
+class ConstanceConfigWrapper(object):
+    """A Constance configuration wrapper to allow overriding the config."""
+    _original_values = []
+
+    def __setattr__(self, attr, value):
+        from constance import config
+        self._original_values.append((attr, getattr(config, attr)))
+        setattr(config, attr, value)
+        # This can fail if Constance uses a cached database backend
+        # CONSTANCE_DATABASE_CACHE_BACKEND = False to disable
+        assert getattr(config, attr) == value
+
+    def finalize(self):
+        from constance import config
+        for attr, value in reversed(self._original_values):
+            setattr(config, attr, value)
+        del self._original_values[:]
+
+
+@pytest.fixture
+def constance_config(db, settings):
```

Comment:
❤️  This fixture and its associated `ConstanceConfigWrapper` class are really beautifully done. I like this fixture so much better than being limited to using `override_config` as a context manager within the new-style tests.

This fixture will be generally useful (for tests within `kuma.attachments`, `kuma.landing`, `kuma.spam`, `kuma.wiki`, etc.) so could be moved now to the top-level `conftest.py` file, but if you'd rather wait until it's used in those places that's fine too.